js/autotrack/textInput.js

```diff
@@ -37,3 +42,57 @@ const debouncedAutocaptureTextInputChange = track => (
 
   track(eventType, autotrackProps);
 };
+
+// :HACK: In previous implementations, there would be a 'TextInput' component somewhere in the hierarchy.  However, with this
+// implementation, this is not the case (even the TextInputComponent has a displayName of 'InternalTextInput').  This means that any event
+// definitions on 'TextInput' would break.
+// To get around this, wrap the returned HOC with a no-op component with the 'TextInput' display name.
+// :TODO: (jmtaber129): Consider other workarounds to this, like setting the display name of instrumented components when they're exported
+// from the React Native lib (via instrumentation).
+const NoopTextInput = props => {
+  return props.children;
+};
+
+NoopTextInput.displayName = 'TextInput';
+
+export const withHeapTextInputAutocapture = track => TextInputComponent => {
+  class HeapTextInputAutocapture extends React.Component {
+    autocaptureTextInputChangeWithDebounce = bailOnError(
+      _.debounce(debouncedAutocaptureTextInputChange(track), DEBOUNCE_PERIOD_MS)
+    );
+
+    render() {
+      const { forwardedRef, onChange, ...rest } = this.props;
+
+      return (
+        <TextInputComponent
+          ref={forwardedRef}
+          onChange={e => {
+            this.autocaptureTextInputChangeWithDebounce('text_edit', this, e);
```

Comment:
would it be worthwhile to move `text_edit` to a constants file (assuming that its used in various other places)

Reply:
These event types are kinda all over the place in the repo (some are in the instrumentor, others are in the library), so I think I'd prefer to leave this sort of thing for another time.